<a href="https://colab.research.google.com/github/jelissad-29/2067-html-css-desarrollo-estructura-archivos/blob/main/AluraStoreLatam.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [7]:
import pandas as pd

# URLs de las bases de datos (Copia este bloque completo)
url1 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

# Cargamos los datos
tienda = pd.read_csv(url1)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

print("¡Listo! Datos cargados sin errores.")

¡Listo! Datos cargados sin errores.


In [8]:
# Sumamos la columna 'Precio' para cada una de las tiendas
fact_t1 = tienda['Precio'].sum()
fact_t2 = tienda2['Precio'].sum()
fact_t3 = tienda3['Precio'].sum()
fact_t4 = tienda4['Precio'].sum()

print(f"Facturación Tienda 1: ${fact_t1:,.2f}")
print(f"Facturación Tienda 2: ${fact_t2:,.2f}")
print(f"Facturación Tienda 3: ${fact_t3:,.2f}")
print(f"Facturación Tienda 4: ${fact_t4:,.2f}")

Facturación Tienda 1: $1,150,880,400.00
Facturación Tienda 2: $1,116,343,500.00
Facturación Tienda 3: $1,098,019,600.00
Facturación Tienda 4: $1,038,375,700.00


In [9]:
# Analizamos la frecuencia de las categorías
tiendas_lista = [tienda, tienda2, tienda3, tienda4]

for i, df in enumerate(tiendas_lista, 1):
    categoria_top = df['Categoría del Producto'].value_counts().idxmax()
    ventas = df['Categoría del Producto'].value_counts().max()
    print(f"Tienda {i} - Categoría líder: {categoria_top} ({ventas} ventas)")

Tienda 1 - Categoría líder: Muebles (465 ventas)
Tienda 2 - Categoría líder: Muebles (442 ventas)
Tienda 3 - Categoría líder: Muebles (499 ventas)
Tienda 4 - Categoría líder: Muebles (480 ventas)


In [10]:
# Calculamos la media de la columna 'Calificación'
for i, df in enumerate(tiendas_lista, 1):
    promedio = df['Calificación'].mean()
    print(f"Tienda {i} - Calificación promedio: {promedio:.2f} estrellas")

Tienda 1 - Calificación promedio: 3.98 estrellas
Tienda 2 - Calificación promedio: 4.04 estrellas
Tienda 3 - Calificación promedio: 4.05 estrellas
Tienda 4 - Calificación promedio: 4.00 estrellas


In [11]:
for i, df in enumerate(tiendas_lista, 1):
    top_prod = df['Producto'].value_counts().idxmax()
    bottom_prod = df['Producto'].value_counts().idxmin()
    print(f"Tienda {i}: [TOP: {top_prod}] | [MENOS VENDIDO: {bottom_prod}]")

Tienda 1: [TOP: Microondas] | [MENOS VENDIDO: Auriculares con micrófono]
Tienda 2: [TOP: Iniciando en programación] | [MENOS VENDIDO: Juego de mesa]
Tienda 3: [TOP: Kit de bancas] | [MENOS VENDIDO: Bloques de construcción]
Tienda 4: [TOP: Cama box] | [MENOS VENDIDO: Guitarra eléctrica]


In [12]:
# Promedio de la columna 'Costo de envío'
for i, df in enumerate(tiendas_lista, 1):
    envio = df['Costo de envío'].mean()
    print(f"Tienda {i} - Costo envío promedio: ${envio:,.2f}")

Tienda 1 - Costo envío promedio: $26,018.61
Tienda 2 - Costo envío promedio: $25,216.24
Tienda 3 - Costo envío promedio: $24,805.68
Tienda 4 - Costo envío promedio: $23,459.46


Informe de Desempeño Estratégico: Alura Store
Analista: Esther Jelissa Grullón Díaz

1. Resumen de Hallazgos
Tras procesar los datos de las cuatro unidades de negocio, se han obtenido los siguientes indicadores clave:

Facturación Total: La Tienda 1 lidera las ventas con $1,150,880,400.00, mientras que la Tienda 4 registra el volumen más bajo con $1,038,375,700.00.

Tendencias de Consumo: La categoría de Muebles es el motor principal de todas las tiendas, destacando la Tienda 3 con 499 ventas en este rubro.

Satisfacción del Cliente: La Tienda 3 posee la mejor reputación (4.05 estrellas), mientras que la Tienda 1 presenta el promedio más bajo (3.98 estrellas).

Eficiencia Logística: La Tienda 4 es la más eficiente en envíos ($23,459.46 por pedido), contrastando con la Tienda 1, que tiene el costo operativo más alto ($26,018.61).

2. Diagnóstico de Productos
Producto Estrella: El producto con mayor tracción es el Kit de bancas (Tienda 3) y la Cama box (Tienda 4).

Puntos Críticos: Los Auriculares con micrófono y las Guitarras eléctricas son los productos con menor rotación, representando capital inmovilizado.

3. Recomendación Final para el Sr. Juan
Basado en el análisis de datos, se recomienda vender la Tienda 4.

Justificación:
Aunque la Tienda 4 es eficiente en sus costos de envío, es la sucursal que menos facturación total aporta al grupo (una diferencia de más de 112 millones respecto a la Tienda 1). Para la estrategia de reinversión del Sr. Juan, liquidar esta unidad proporcionará el capital necesario minimizando el impacto en el flujo de caja global de Alura Store.

Se sugiere además que el capital obtenido se reinvierta en potenciar la categoría de Muebles en las tiendas restantes, ya que es el segmento con mayor éxito probado.